### PR TITLE
Support eth_getBalance in mapping

### LINKED ISSUE #895: Support for calling Ether Balances in mappings
Right now there is no way to call an ether balance of an account or a contract. 

This functionality is very useful in protocols that use ether - this includes Uniswap and Compound, and likely also Dharma and Maker. 

This might block me on Uniswap (figuring out if I can work around it right now). With compound we were able to get around it, because they added a getter function that allowed us to read the ether balance of their smart contracts. 

It should return the Eth balance in WEI.

### PR DESCRIPTION
**Pull request by incrypto32**
Closes #895

### CODE REVIEW

File: chain/ethereum/src/runtime/runtime_adapter.rs [hunk 1]
```diff
@@ -132,6 +147,35 @@ fn ethereum_call(
     }
 }
 
+fn eth_get_balance(
+    eth_adapter: &EthereumAdapter,
+    ctx: HostFnCtx<'_>,
+    wasm_ptr: u32,
+) -> Result<AscPtr<AscBigInt>, HostExportError> {
```
Comment (lutter) [inline chain/ethereum/src/runtime/runtime_adapter.rs:154]: Do we need to charge gas for this call? I am not sure about the mechanics/reasoning of gas, but it feels wrong to make this free.

**Review by lutter — APPROVED** [review]

**Comment by lutter** [comment]
For the failing unit tests, if you rebase onto latest master, the tests should pass.

**Comment by lutter** [comment]
Oh, one more thing: do we need a new spec or API version because this adds a host fn?

File: chain/ethereum/src/runtime/runtime_adapter.rs (hunk 1 above)
Comment (incrypto32) [reply chain/ethereum/src/runtime/runtime_adapter.rs:154]: Right, this shouldn't be gas free. @leoyvens how would we determine a good gas value for `eth_getBalance`?

Comment (leoyvens) [reply chain/ethereum/src/runtime/runtime_adapter.rs:155]: I suppose we can start with charging the same we charge for `eth_call`.

**Pull request merged**